lib/elixir/lib/kernel/parallel_compiler.ex

```diff
@@ -355,32 +355,36 @@ defmodule Kernel.ParallelCompiler do
   defp each_cycle_return(other), do: other
 
   defp write_and_verify_modules(result, warnings, dependent_modules, state) do
-    %{output: output, beam_timestamp: beam_timestamp} = state
+    write_module_binaries(result, state)
```

Comment:
Have this function return the module names so we don't need to traverse the result once again downstream. :)